### PR TITLE
interchainstaking: export duplicating errors

### PR DESCRIPTION
**Pull request by pysel**
- **export**
- **export**

## 1. Summary
Fixes # (issue)

## 2.Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## 5. Checklist

- [ ] Does the Readme need to be updated?

## Summary by CodeRabbit

- **Bug Fixes**
  - Improved error handling by replacing generic error messages with specific custom error types for better clarity and reporting across multiple functions.
  
- **New Features**
  - Introduced new custom error types (`ErrNoRegisteredZoneForChainId`, `ErrEmptyChainID`, `ErrErrorHappened`, `ErrUnexpectedNegative`) to provide more specific error information.

- **Refactor**
  - Updated various functions to adopt the new custom error types for more precise error messaging and handling.

### CODE REVIEW

File: x/interchainstaking/keeper/grpc_query.go [hunk 1]
```diff
@@ -116,7 +116,7 @@ func (k *Keeper) DepositAccount(c context.Context, req *types.QueryDepositAccoun
 
 	zone, found := k.GetZone(ctx, req.GetChainId())
 	if !found {
-		return nil, status.Error(codes.NotFound, fmt.Sprintf("no zone found matching %s", req.GetChainId()))
```
Comment (joe-bowman) [inline x/interchainstaking/keeper/grpc_query.go:119]: Are the status errors, returning codes linked to the HTTP response code? If so, do these need to remain wrapped?

Comment (pysel) [reply x/interchainstaking/keeper/grpc_query.go]: this might be true, good catch. I reverted changes of status'ed errors, should be good to merge now

**Comment by joe-bowman** [comment]
Can you fix the lint errors please (I think it's Id -> ID). Then we will get this merged. Thanks!

**Comment by joe-bowman** [comment]
This pull request has been automatically marked as stale because it has not had any recent activity. It will be closed if no further activity occurs. Thank you!

**Comment by joe-bowman** [comment]
This pull request has been automatically marked as stale because it has not had any recent activity. It will be closed if no further activity occurs. Thank you!

**Comment by joe-bowman** [comment]
This pull request has been automatically marked as stale because it has not had any recent activity. It will be closed if no further activity occurs. Thank you!

**Pull request closed without merging**